libs/model-assessment/src/lib/ModelAssessmentDashboard/Cohort/ProcessPreBuiltCohort.ts

```diff
@@ -0,0 +1,235 @@
+  ) {
+    const modelClasses = jointDataset.getModelClasses();
+    const index: number[] = [];
+    for (const modelClass of preBuiltCohortFilter.arg) {
+      const indexModelClass = modelClasses.indexOf(modelClass);
+
+      if (indexModelClass !== -1) {
+        index.push(indexModelClass);
+      }
+    }
+
+    index.sort((a, b) => a - b);
+
+    const filter: IFilter = {
+      arg: index,
+      column: filterColumnName,
+      method: preBuiltCohortFilter.method
+    } as IFilter;
+    return filter;
+  }
+  const filter: IFilter = {
+    arg: preBuiltCohortFilter.arg,
+    column: filterColumnName,
+    method: preBuiltCohortFilter.method
+  } as IFilter;
+  return filter;
+}
+
+function translatePreBuiltCohortFilterForClassificationOutcome(
+  preBuiltCohortFilter: IPreBuiltFilter,
+  jointDataset: JointDataset
+): IFilter {
+  const index: number[] = [];
+  if (jointDataset.metaDict[JointDataset.ClassificationError]) {
+    const allowedCalssificationErrorValues =
+      jointDataset.metaDict[JointDataset.ClassificationError]
+        .sortedCategoricalValues;
+
+    if (allowedCalssificationErrorValues !== undefined) {
+      for (const classificationError of preBuiltCohortFilter.arg) {
+        const indexclassificationError =
+          allowedCalssificationErrorValues.indexOf(classificationError);
+
+        if (indexclassificationError !== -1) {
+          index.push(indexclassificationError);
+        }
+      }
+    }
+  }
+  index.sort((a, b) => a - b);
+  const filter: IFilter = {
+    arg: index,
+    column: JointDataset.ClassificationError,
+    method: preBuiltCohortFilter.method
+  } as IFilter;
+  return filter;
+}
+
+function translatePreBuiltCohortFilterForDataset(
+  preBuiltCohortFilter: IPreBuiltFilter,
+  jointDataset: JointDataset
+): [IFilter | undefined, string | undefined] {
+  let jointDatasetFeatureName = undefined;
+  let userDatasetFeatureName = undefined;
+  for (jointDatasetFeatureName in jointDataset.metaDict) {
+    if (
+      jointDataset.metaDict[jointDatasetFeatureName].abbridgedLabel ===
+      preBuiltCohortFilter.column
+    ) {
+      userDatasetFeatureName =
+        jointDataset.metaDict[jointDatasetFeatureName].abbridgedLabel;
+      break;
+    }
+  }
+
+  if (
+    jointDatasetFeatureName === undefined ||
+    userDatasetFeatureName === undefined
+  ) {
+    return [undefined, localization.Core.PreBuiltCohort.featureNameNotFound];
+  }
+
+  if (preBuiltCohortFilter.method === FilterMethods.Includes) {
+    if (!jointDataset.metaDict[jointDatasetFeatureName].isCategorical) {
+      return [
+        undefined,
+        localization.Core.PreBuiltCohort.notACategoricalFeature
+      ];
+    }
+    const index: number[] = [];
+    const categorcialValues =
+      jointDataset.metaDict[jointDatasetFeatureName].sortedCategoricalValues;
+    if (categorcialValues !== undefined) {
+      for (const categoricalValue of preBuiltCohortFilter.arg) {
```

Comment:
this loop looks similar to above, do you think this might be refactored with above code into a method?